prov/gni/include/gnix.h

```diff
@@ -391,16 +392,13 @@ struct gnix_fid_ep {
 	struct gnix_tag_storage tagged_unexp_recv_queue;
 	struct gnix_tag_storage tagged_posted_recv_queue;
 
-
-	fastlock_t recv_comp_lock;
-	struct slist pending_recv_comp_queue;
 	/* pointer to tag matching engine */
-	void *tag_matcher;
```

Comment:
Is the removal of these fields general clean up?  Perhaps they were put in for future use?  Not that I think they should be maintained, but if there's some useful background info, we should document it in the commit message.